module/attribute/src/Application/Controller/Api/Attribute/AttributeAutocompleteAction.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\Attribute\Application\Controller\Api\Attribute;
+
+use Ergonode\Api\Application\Response\SuccessResponse;
+use Ergonode\Attribute\Domain\Query\AttributeQueryInterface;
+use Ergonode\Core\Domain\ValueObject\Language;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Swagger\Annotations as SWG;
+
+/**
+ * @Route("attribute/autocomplete", methods={"GET"})
+ */
+class AttributeAutocompleteAction
+{
+    /**
+     * @var AttributeQueryInterface
+     */
+    private AttributeQueryInterface $attributeQuery;
+
+    /**
+     * @param AttributeQueryInterface $attributeQuery
+     */
+    public function __construct(AttributeQueryInterface $attributeQuery)
+    {
+        $this->attributeQuery = $attributeQuery;
+    }
+
+    /**
+     * @SWG\Tag(name="Attribute")
+     * @SWG\Parameter(
+     *     name="language",
+     *     in="path",
+     *     type="string",
+     *     required=true,
+     *     default="en_GB",
+     *     description="Language Code",
+     * )
+     * @SWG\Parameter(
+     *     name="search",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     description="searched value"
+     * )
+     * @SWG\Parameter(
+     *     name="limit",
+     *     in="query",
+     *     required=false,
+     *     type="integer",
+     *     description="searched value count"
+     * )
+     * @SWG\Parameter(
+     *     name="field",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     description="Order field",
+     * )
+     * @SWG\Parameter(
+     *     name="order",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     enum={"ASC","DESC"},
+     *     description="Order",
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns attributes",
+     * )
+     *
+     * @param Language $language
+     * @param Request  $request
+     *
+     * @return Response
+     */
+    public function __invoke(Language $language, Request $request): Response
+    {
+        $search = $request->query->get('search');
+        $limit = $request->query->getInt('limit', null);
```

Comment:
Will it work properly if one provides non-numeric string?

Reply:
yes ;)